app/src/test/java/com/droidhats/campuscompass/viewmodels/MapViewModelTest.kt

```diff
@@ -13,15 +15,48 @@ import org.robolectric.annotation.Config
 @Config(sdk = [Build.VERSION_CODES.O_MR1])
 class MapViewModelTest {
     private val viewmodel: MapViewModel = MapViewModel(RuntimeEnvironment.application)
+    private lateinit var campuses: List<Campus>
+    private lateinit var sgw: String
+    private lateinit var loyola: String
+    private lateinit var selectedBuilding: Building
 
-    @Test
-    fun testInit() {
-        val campuses: List<Campus> = viewmodel.getCampuses()
-        val sgw = "SGW"
-        val loyola = "Loyola"
+    @Before
+    fun setup(){
+        campuses = viewmodel.getCampuses()
+        sgw = "SGW"
+        loyola = "Loyola"
+    }
 
+    @Test
+    fun initTest() {
         // Assert that the proper campuses are there
         Assert.assertEquals(sgw, campuses[0].name)
         Assert.assertEquals(loyola, campuses[1].name)
     }
+
+    @Test
+    fun getNavigationRepositoryTest(){
+        val navigationRepository = viewmodel.navigationRepository
+        Assert.assertNotNull(navigationRepository)
+    }
+
+    @Test
+    fun findBuildingByPolygonTagTest_BuildingWithMarker(){
+        //Extracting Hall building
+        var expectedBuilding = campuses[0].getBuildings()[0]
+
+        selectedBuilding = viewmodel.findBuildingByPolygonTag("Henry F. Hall Building")!!
+        Assert.assertEquals(expectedBuilding, selectedBuilding)
+    }
+
+    @Test
+    fun findBuildingByPolygonTagTest_NullMarker(){
+        //Extracting RR Annex
+        var expectedBuilding = campuses[0].getBuildings()[11]
```

Comment:
Change this to val please.

Since expectedBuiding is not being modified but just being used you can safely change this to val (immutable declaration) as mentioned here: https://kotlinlang.org/docs/reference/properties.html

Reply:
Resolved this comment and all of the above with 
4a91644
edbd9aa